rancher2/schema_cluster_v2_rke_config_system_config.go

```diff
@@ -51,6 +54,47 @@ func clusterV2RKEConfigSystemConfigLabelSelectorFields() map[string]*schema.Sche
 	return s
 }
 
+func clusterV2RKEConfigSystemConfigFieldsV0() map[string]*schema.Schema {
+	s := map[string]*schema.Schema{
```

Comment:
I do not know how TF works, so I can be wrong here.

Ain't this function exactly the same as  clusterV2RKEConfigSystemConfigFields() ?

If names are important,  shouldn't u have a base function and wrap it ?

This is the same for clusterV2RKEConfigFieldsV0() and clusterV2FieldsV0()

Reply:
Good question :) yes, they are the same but there need to be 2 schema definitions for the state migration of migration `machine_selector_config.config` from a `Type.Map` to a `Type.String` that supports yaml. That way the top level RKE Config schema will reference the correct types both on cluster creation and on an upgrade of the provider. Check out the docs on [Terraform v0.12 State Migrations](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/state-migration) if you are curious